metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -2624,4 +2693,101 @@ class MetalsLanguageServer(
     )
   }
 
+  private def isMillBuildSc(path: AbsolutePath): Boolean =
+    path.toNIO.getFileName.toString == "build.sc" &&
+      // for now, this only checks for build.sc, but this could be made more strict in the future
+      // (require ./mill or ./.mill-version)
+      buildTools.isMill
+
+  def maybeImportScript(path: AbsolutePath): Option[Future[Unit]] = {
+    val directory = path.parent
+    if (
+      ammonite.loaded(path) || scalaCli.loaded(directory) || isMillBuildSc(path)
+    )
+      None
+    else {
+      def doImportScalaCli(): Unit =
+        scalaCli.start(scalaCli.roots :+ directory).onComplete {
+          case Failure(e) =>
+            languageClient.showMessage(
+              Messages.ImportScalaScript.ImportFailed(path.toString)
+            )
+            scribe.warn(s"Error importing Scala CLI project $directory", e)
+          case Success(_) =>
+            languageClient.showMessage(
+              Messages.ImportScalaScript.ImportedScalaCli
+            )
+        }
+      def doImportAmmonite(): Unit =
+        ammonite.start(Some(path)).onComplete {
+          case Failure(e) =>
+            languageClient.showMessage(
+              Messages.ImportScalaScript.ImportFailed(path.toString)
+            )
+            scribe.warn(s"Error importing Ammonite script $path", e)
+          case Success(_) =>
+            languageClient.showMessage(
+              Messages.ImportScalaScript.ImportedAmmonite
+            )
+        }
+
+      val autoImportAmmonite =
+        tables.dismissedNotifications.AmmoniteImportAuto.isDismissed
```

Comment:
We should probably create a separate table for this instead of operating on the dismissed status of notification.

Reply:
Why? That's a "dismiss-able" notification too.